src/v/iceberg/rest_client/entities.cc

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "iceberg/rest_client/entities.h"
+
+#include <boost/algorithm/string/join.hpp>
+
+namespace iceberg::rest_client {
+
+table::table(
+  std::string_view root_url, const chunked_vector<ss::sstring>& namespace_parts)
+  : rest_entity(root_url)
+  , _namespace{boost::algorithm::join(namespace_parts, "\x1f")} {}
```

Comment:
Note: I looked into `fmt::join` here but it seems to return something not immediately convertible to a seastar string.